### PR TITLE
Fix issues with Addon Auto-publishing

### LINKED ISSUE #2231: Can't publish Addon config, when there are no `publishables`
## Bug Description

Statamic registers an addon `tag` even if there are no publishables

```
        $publishables = collect($this->publishables)
            ->mapWithKeys(function ($destination, $origin) use ($package) {
                return [$origin => public_path("vendor/{$package}/{$destination}")];
            });

        $this->publishes($publishables->all(), $this->getAddon()->slug());
```

In the above code, a tag is created even if `$publishables` is empty, and on `composer dump-autoload` when Statamic tries to publish, it uses the tag NOT the Serviceprovider.

## How to Reproduce

Create an addon that publishes a config, but no `$publishables`, then when installing you'll see an error.

## Environment

Statamic 3.0.0-beta.44 Pro
Laravel 7.25.0
PHP 7.4.9

**Install method (choose one):**
- Fresh install from `statamic/statamic`

### PR DESCRIPTION
**Pull request by damcclean**
For the most part, addon auto-publishing works fine, there's just two things I've fixed in this pull request:

* when an addon has no publishable assets, Statamic shouldn't register publishable stuff (it was doing `$this->publishes` even if there was nothing needing published)
* it also sorts an issue where it would try and publish an addon's stuff even if a tag didn't exist or there wasn't anything to publish. Even after the last thing was fixed, it still wouldn't solve this issue:

image

This PR fixes #2231. Hopefully this pull request is helpful.

### CODE REVIEW

File: src/Providers/AddonServiceProvider.php [hunk 1]
```diff
@@ -170,7 +170,9 @@ protected function bootPublishables()
                 return [$origin => public_path("vendor/{$package}/{$destination}")];
             });
 
-        $this->publishes($publishables->all(), $this->getAddon()->slug());
+        if ($publishables->count() >= 1) {}
```
Comment (edalzell) [inline src/Providers/AddonServiceProvider.php]: if (! $publishables->isNotEmpty())

**Comment by jasonvarga** [comment]
What's the reason for the new hasPublishables property?

**Comment by damcclean** [comment]
> What's the reason for the new hasPublishables property?

That's a good question, it would probably work with the existing one. I'll switch it.

**Comment by damcclean** [comment]
> What's the reason for the new hasPublishables property?

It's so that if an addon doesn't have any publishable, it doesn't try and publish it. 

Although maybe it would be a good idea just to set `publishAfterInstall` to false if there's nothing instead of that new property.

**Comment by damcclean** [comment]
@jasonvarga I've removed the `$hasPublishables` property from the pull request.

**Comment by damcclean** [comment]
> I still don’t think it should disable it like that. We still want to publish if there are scripts or style sheets.

How do you suggest doing it then to avoid the `Unable to locate publishable resources` error popping up?

**Pull request closed without merging**